app/_src/gateway/plugin-development/get-started/testing.md

```diff
@@ -0,0 +1,360 @@
+}
+```
+
+With the test environment initialized lets see how to manually run our plugin code. 
+
+### 3. Manually test plugin
+
+Our plugin is installed, let's configure {{site.base_gateway}} entities so we can invoke and validate 
+our plugins behavior.
+
+{:.note}
+> **Note**: For each of the following `POST` requests to the Admin API, you should receive
+> a `HTTP/1.1 201 Created` response from {{site.base_gateway}} indicating successful creation of the entity.
+
+Still within the {{site.base_gateway}} container's shell, add a new service with the following:
+
+```sh
+curl -i -s -X POST http://localhost:8001/services \
+    --data name=example_service \
+    --data url='http://httpbin.org'
+```
+
+Associate our custom plugin with the `example_service` service:
+
+```sh
+curl -is -X POST http://localhost:8001/services/example_service/plugins \
+    --data 'name=my-plugin'
+```
+
+Add a new route so we can send requests through the `example_service`:
+
+```sh
+curl -i -X POST http://localhost:8001/services/example_service/routes \
+    --data 'paths[]=/mock' \
+    --data name=example_route
+```
+
+Our plugin is now configured to be invoked when {{site.base_gateway}} proxies
+requests via the `example_service`. Prior to forwarding the response from the 
+upstream, our plugin should append the `X-MyPlugin` header to the list of response headers.
+
+Let's test the behavior by proxying a request and asking `curl` to show the 
+response headers with the `-i` flag:
+
+```sh
+curl -i http://localhost:8000/mock/anything
+```
+
+`curl` should report `HTTP/1.1 200 OK` and show us the response headers from the gateway. Included
+in the set of headers, should be `X-MyPlugin: response`, indicating that our plugin logic has been invoked.
+
+For example: 
+
+```sh
+HTTP/1.1 200 OK
+Content-Type: application/json
+Connection: keep-alive
+Content-Length: 529
+Access-Control-Allow-Credentials: true
+Date: Tue, 12 Mar 2024 14:44:22 GMT
+Access-Control-Allow-Origin: *
+Server: gunicorn/19.9.0
+X-MyPlugin: response
+X-Kong-Upstream-Latency: 97
+X-Kong-Proxy-Latency: 1
+Via: kong/3.6.1
+X-Kong-Request-Id: 8ab8c32c4782536592994514b6dadf55
+```
+
+Exit the {{site.base_gateway}} shell before proceeding:
+
+```sh
+exit
+```
+
+For quickly getting started, manually validating a plugin using the Pongo shell works
+nicely. However, you will prefer to deploy automated testing and maybe a Test-driven development 
+(TDD) methodology. Let's see how Pongo can help with this as well.
+
+### 4. Write a test
+
+Pongo supports running automated tests using the 
+[Busted](https://lunarmodules.github.io/busted/) Lua test framework. In plugin
+projects, the test files reside under the `spec/<plugin-name>` directory. For this project
+we created the `spec/my-plugin` folder earlier.  
+ 
+The following is a code listing for a test that validates our plugin's current behavior. 
+Copy this code and place it into a new file located at `spec/my-plugin/01-integration_spec.lua`. 
```

Comment:
For parallelism, there should be a `touch` command here first:

```
touch spec/my-plugin/01-integration_spec.lua
```
(can't add as GH suggestion due to the backticks)

Reply:
I don't agree with this one.  The reason for the `touch` commands in the setup section was that I wanted to teach the user about required modules and the "setup" of a new project. Having them touch the files first helped show required files.  I don't think having the user `touch` files here and then open an editor and paste is necessary here.